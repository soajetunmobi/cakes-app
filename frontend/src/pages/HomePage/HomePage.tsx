import React, { FunctionComponent } from 'react'
import { useHistory } from 'react-router-dom'
import { LoadingIndicator } from '../../components/LoadingIndicator'
import { ErrorPanel } from '../../components/ErrorPanel'
import { routes } from '../../common/const/routes'
import { useFetchCakes } from '../../hooks'

const ADD_BUTTON_LABEL = 'Add comment'
const VIEW_BUTTON_LABEL = 'View comment'

export const HomePage: FunctionComponent = () => {
  const history = useHistory()

  const { error, isLoading, cakes } = useFetchCakes()

  if (isLoading || !cakes) {
    return <LoadingIndicator />
  }

  if (error) {
    const errorMessage = <div>{error}</div>
    return <ErrorPanel children={errorMessage} />
  }

  const renderCakes = () => {
    return (
      cakes &&
      cakes.map(cake => (
        <div className='cake cursor--pointer' key={cake.id}>
          <span className='cake__name'>{cake.name}</span>
          <span className='cake__image'>
            <img alt={cake.name} src={cake.imageUrl} />
          </span>
          <div className='buttons-container buttons-container--flex-end'>
            <button className='button--secondary' onClick={() => viewReview(cake.id)}>
              {VIEW_BUTTON_LABEL}
            </button>
          </div>
        </div>
      ))
    )
  }

  const addCake = () => {
    history.push(routes.add)
  }

  const viewReview = (id: number) => {
    const path = {
      pathname: `/cakes/${id}/review`,
      state: { id }
    }
    history.push(path)
  }

  return (
    <div>
      <h2 className='page__title'>Cakes</h2>
      <div className='cakes'>{cakes && renderCakes()}</div>
      <div className='buttons-container buttons-container--center margin--bottom--double'>
        <button className='button--primary button--w250' onClick={() => addCake()}>
          {ADD_BUTTON_LABEL}
        </button>
      </div>
    </div>
  )
}
