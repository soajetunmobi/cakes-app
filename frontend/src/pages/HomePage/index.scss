.cakes{
  display: grid;
  grid-template-columns: repeat(1, 1fr);
  grid-template-rows: 1fr 1fr;

  @include respond-to('medium') {
    grid-template-columns: repeat(2, 1fr);
    grid-template-rows: 1fr 1fr;
  }

  @include respond-to('large') {
    grid-template-columns: repeat(3, 1fr);
    grid-template-rows: 1fr 1fr;
  }
}

.cake {
  border: 1px solid #dee2e6;
  border-radius: .45rem;
  display: flex;
  flex-direction: column;
  padding: px-to-rem(15px);
  margin: px-to-rem(15px);

  &__image {
    display: block;
    max-width: 400px;
    width: 100%;
    margin: auto;

    @include respond-to('medium') {
      width: 400px;
    }

    & > img {
      width: 100%;
    }
  }

  &__name {
    font-size: 18px;
    margin-bottom: 10px;
    text-align: center;
  }
}

