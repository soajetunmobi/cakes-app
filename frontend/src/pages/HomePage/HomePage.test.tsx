import React from 'react'
import renderer from 'react-test-renderer'
import { HomePage } from './HomePage'

describe('Home page component', () => {
  test('it renders correctly', () => {
    const root = renderer.create(<HomePage />).toJSON()
    expect(root).toMatchSnapshot()
  })
})
