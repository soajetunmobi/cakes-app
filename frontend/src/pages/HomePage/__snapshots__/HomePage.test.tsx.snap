// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page component it renders correctly 1`] = `
<div
  className="loader"
/>
`;
